### ISSUE TITLE
get configfile and result in plugin

### ISSUE THREAD

**Issue by githuber0007** [eslint/eslint#16237]
### ESLint version

v8.22.0

### What problem do you want to solve?

```javascript
module.exports = {
  ...
  // would be great if create can return a Promise
  create: function(context) {
      return {
          // get messages in context
          // get eslintrc in context
      };
  }
```

1. I'm making an eslint plugin, the lint result of the plugin's rules may conflict with other rules, but in the existing context API I can't know which rule my plugin conflicts with. If I know, I can block the rules in this plugin that conflict with other rules in next loop.
2. Also I want to get the eslintrc configuration object in the plugin's create scope, there is still no way to get the eslintrc configuration in the current cwd directly from the context API.
3. I solved the above two problems temporarily by re-creating an instance of ESlint in the plugin, but running another instance of ESLint in one instance is always superfluous. 

Hope to hear back.

### What do you think is the correct solution?

I would like to add to the context API an object that collects all reports, similar to the messages Array in result , and the ability to access the eslintrc configuration object that has been read in the ESlint instance.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Rules intentionally do not have access to other rules. Rules are supposed to be standalone and not have dependencies on the results of other rules.

Similarly, rules are only supposed to care about their own configuration, not anything else. Everything that is safe for a rule to access it already passed on the context object.

This was an intentional design decision in ESLint because of rules can react based on other rules, it’s nearly impossible to figure out what’s going wrong.

**Comment by githuber0007** [comment]
I know this is the original intention of your design, but we need a way to know if our plugin conflicts with other plugins. Otherwise, how can I make sure that my plugin rules don't conflict with the rules of other massive ESLint community plugins. It is obviously impossible to test one by one.

I think this is a very realistic scenario, and unless ESLint mandates that only one plugin can be enabled at a time, this can make the results of using multiple plugins undesired and unstable, and the process will Very time consuming. Leaving the user to analyze and disable conflicting rules should not be the best option.

With the increase in the number of ESLint community plug-in rules, one day new plug-in development will have to consider the former rules, old rules maintenance will have to consider the rules of newly added plug-ins, and the coordination between the rules of different ESLint community developers Cooperation is a foreseeable need. The forced incompatibilities between the rules of different developers will eventually limit the growth of ESLint's plugin community.
